### PR TITLE
General reward index

### PR DESCRIPTION
**Pull request by madninja**
The reward-index (configured for `iot` or `mobile` is run to index the total accumulated reward shares for hotspots. The accumulation is done from reward shares for the configured network, after a given start time encoded by the `last_reward_manifest` timestamp.

### CODE REVIEW

**Comment by bbalser** [comment]
Is there a reason we need to run 2 instances (mobile, iot) of the indexer instead of a single instance that does both mobile and iot?

**Review by bbalser — APPROVED** [review]

**Comment by madninja** [comment]
> Is there a reason we need to run 2 instances (mobile, iot) of the indexer instead of a single instance that does both mobile and iot?

The two networks are independent as are the reward tokens earned by each. Cleaner separation of concerns while still sharing common code. This way they operate on their own buckets, their own dbs in their own environment

**Review by jeffgrunewald — APPROVED** [review]

File: reward_index/src/indexer.rs [hunk 1]
```diff
@@ -14,26 +17,28 @@ pub struct Indexer {
...
+            mode: settings.mode,
             interval: settings.interval(),
             verifier_store: FileStore::from_settings(&settings.verifier).await?,
             pool,
         })
     }
 
     pub async fn run(&mut self, shutdown: triggered::Listener) -> Result<()> {
-        tracing::info!("starting mobile index");
+        tracing::info!(mode = self.mode.to_string(), "starting index");
```
[the remark this answers is not in the document]
Comment (madninja) [reply reward_index/src/indexer.rs:35]: They’re all key value pairs. The single string is a shortcut for “message = …”

[the remark this answers is not in the document]
Comment (jeffgrunewald) [reply reward_index/src/indexer.rs:35]: wow, TIL

**Pull request merged**